Sprint 1

DeLonghi opened the pull request.

chinarev committed:
add ShopCatalog class

01gaM committed:
WineShop class added

+ object info to logs method
+ object info to file method

01gaM committed:
Merge branch 'wine-shop-class' into sprint-1

chinarev committed:
small fix shopCatalog

Murwar committed:
Created db Tables

Creation tables for testing controller connection

Murwar committed:
Sprint1_Classes and Controller Creation

Add Brands, Countries, Grapes, WineGrapesInfo, Wine Classes. The controller has been created. Controller Connection tested through swagger-UI.

DeLonghi committed:
Merge pull request #3 from chinarev/ShopCatalog

Shop Catalog

DeLonghi committed:
Merge pull request #1 from 01gaM/wine-shop-class

WineShop class added

Murwar committed:
Create simple Wine Class

DeLonghi committed:
Merge pull request #2 from Murwar/sprint-1

Sprint 1_Classes, Controller and Tables creation

01gaM committed:
Merge pull request #1 from microservices-course-itmo/sprint-1

Sync Sprint 1

01gaM committed:
Merge remote-tracking branch 'origin/sprint-1' into sprint-1

01gaM committed:
WineShop junit tests added

+ SimpleParserTest folder created
+ WineShop infoToFile() method changed (requires File object now)

01gaM committed:
Merge branch 'wine-shop-test' into sprint-1

DeLonghi committed:
Merge pull request #4 from 01gaM/wine-shop-test

Wine shop test

01gaM committed:
Merge pull request #2 from microservices-course-itmo/sprint-1

Sync sprint-1 (Merge pull request #4 from 01gaM/wine-shop-test)

Murwar committed:
Fix Flyway Version and DB Table Connection

chinarev committed:
Merge pull request #1 from microservices-course-itmo/sprint-1

Sprint 1

chinarev committed:
update ShopCatalog

DeLonghi committed:
Merge pull request #6 from chinarev/ShopCatalog

UPD shopCatalog

DeLonghi committed:
Merge pull request #5 from Murwar/sprint-1

Fix Flyway Version and DB Table Connection

01gaM committed:
Merge pull request #3 from microservices-course-itmo/sprint-1

Sync with new fixes

DeLonghi committed:
Parser skeleton

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/ShopCatalog.java:
```diff
@@ -0,0 +1,86 @@
+
+    ArrayList<WineShop> shops = new ArrayList<>();
+
+    public int getWineID() {
```
Попробуйте почитать про lombok, это очень удобная штука, которая уже есть в проекте
https://projectlombok.org/
Это поможет вам сократить количество бойлерплейта

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+    public Parser() throws IOException {
+        wineURLs = new ArrayList<String>();
+        WineCatalog = new ArrayList<Wine>();
+        // logger = LoggerFactory.getLogger("ParserLogger");
```
Логгер можно заинжектить с помощью аннотации slf4j

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        float volume = 0;
+        String price = "";
+        int year = 0;
+        float abv = 0;
```
Лучше использовать более информативные имена переменных
Плюс это не самый хороший пример создания переменных в самом верху функции. Их можно создавать по мере их надобности.

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/ShopCatalog.java:
```diff
@@ -0,0 +1,86 @@
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }finally{
```
Неправильное форматирование, используйте ctrl + alt + L перед созданием коммита.

Anglopol commented on pom.xml:
```diff
@@ -32,6 +32,12 @@
+        <dependency>
+             <groupId>org.jsoup</groupId>
+            <artifactId>jsoup</artifactId>
+              <version>1.13.1</version>
```
Перенесите версию в проперти, пожалуйста

Anglopol commented on pom.xml:
```diff
@@ -82,5 +88,10 @@
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Это уже есть в spring test
Можно удалить :)

Anglopol commented on .factorypath:
```diff
@@ -1,4 +1,18 @@
 <factorypath>
+    <factorypathentry kind="VARJAR" id="M2_REPO/com/wine/to/up/service-common-lib/0.2.1-SNAPSHOT/service-common-lib-0.2.1-SNAPSHOT.jar" enabled="true" runInBatchMode="false"/>
```
Это что такое?

shinehlina replied:
Вы добавили же этот файл в gitignore. Тогда что он здесь делает?)

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+
+public class Parser {
+    private final String URL = "https://simplewine.ru";
+    private final int PAGES_TO_PARSE = 21; // < 132
```
Неинформативный комментарий

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+    private ArrayList<String> wineURLs;
+    private ArrayList<Wine> WineCatalog;
+    // private Logger logger;
+    int n_pages;
```
Именование переменной нужно проводить в camelCase

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+    // private Logger logger;
+    int n_pages;
+
+    public Parser() throws IOException {
```
Откуда Exception в конструкторе? Вы же вроде просто два списка создаете

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        String[] name_year;
+
+        doc = Jsoup.connect(URL + "/catalog/vino/").get();
+        int temp = doc.getElementsByAttributeValue("class", "pagination__navigation").size();
```
Лучше перенести весь хардкод в константы, назвать эти константы как-нибудь осмысленно

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/ShopCatalog.java:
```diff
@@ -0,0 +1,86 @@
+            e.printStackTrace();
+        }finally{
+            try {
+                assert fr != null;
```
Откажитесь от ассертов, они проверяются только при запуске программы со специальным ключом. Не оень хорошая практика их использовать

Anglopol commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/ShopCatalog.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @Override
+    public String toString() {
```
lombok генерит toString

Anglopol reviewed (changes requested):
Очень хорошо, комментарии скорее больше по оформлению и использованию некоторых фич

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParserTest/WineShopTest.java:
```diff
@@ -0,0 +1,54 @@
+import java.nio.file.Files;
+
+
+public class WineShopTest {
```
И еще мувните тесты. Они должны быть в папке src/test/java (не в мейне)

shinehlina replied:
Проверьте, что при mvn clean install они запускаются и проходят

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParserTest/WineShopTest.java:
```diff
@@ -0,0 +1,54 @@
+
+    @After
+    public void afterTest() {
+        wineShop = null;
```
Это зачем?)

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParserTest/WineShopTest.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
```
Вот это киньте выше, где поля объявляете

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/controller/MainController.java:
```diff
@@ -0,0 +1,139 @@
+    private WineRepository wineRepository;
+
+
+    @PostMapping(path="/addGrape")
```
Path не должен содержать глаголов. "grape"

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/controller/MainController.java:
```diff
@@ -0,0 +1,139 @@
+
+
+    @PostMapping(path="/addGrape")
+    public @ResponseBody String addGrape(@RequestParam String grapeName){
```
Ставьте это над методом (как PostMapping)

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/domain/entity/Brands.java:
```diff
@@ -0,0 +1,37 @@
+    @Column(name = "brandid")
+    private UUID brandID = UUID.randomUUID();
+
+    @Column(name = "brandname")
```
CamelCase

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/domain/entity/Brands.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @SneakyThrows
+    public void writeInfoToFile(Brands someBrand) {
```
Не думаю, что это прямо в энтити надо делать

01gaM committed:
Requested changes for sprint 1

WineShop class updated:
+ lombok annotations added
+ infoToFile() method changed

Test related updates:
+ WineShopTest moved to test folder
+ New test added
+ JUnit dependency removed from pom.xml

Murwar committed:
changes due to review and some code improvement

DeLonghi committed:
Merge pull request #8 from 01gaM/sprint-1

Requested changes for Sprint 1

DeLonghi committed:
Merge pull request #9 from Murwar/sprint-1

Changes due to review and some code improvement

DeLonghi committed:
Sprint-1 review fix

chinarev committed:
Update ShopCatalog: now with using lombok
+update function with writing into file

DeLonghi committed:
Merge pull request #10 from chinarev/ShopCatalog

Update ShopCatalog

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Wine.java:
```diff
@@ -1,5 +1,55 @@
+            writer = new FileWriter("Wine.txt", true);
+            writer.write(this.toString() + "\n");
+            writer.flush();
+        } catch (IOException e) {
```
Вы где-то этот файл вообще закрываете? Можете использовать try with resources

shinehlina commented on src/main/java/com/wine/to/up/simple/parser/service/SimpleParser/Wine.java:
```diff
@@ -1,5 +1,55 @@
+
+    @Override
+    public String toString() {
+        return "{name=" + this.name + ", brandID=" + this.brandID + ", countryID=" + this.countryID + ", price="
```
Если вы делаете JSON из класса, то используйте конвертеры

shinehlina replied:
(Можете исправить это в следующем pr)

shinehlina commented on src/main/resources/application.properties:
```diff
@@ -13,4 +13,8 @@ spring.datasource.password=postgres
 
+#If you have problems with checksum, and 'flyway:repair' doesn't help,
+#remove flyway_schema_history from DB, remove the comment from the next line, recompile the project with 'mvn clean install -s settings.xml'.
+spring.flyway.baseline-on-migrate=true
```
В идеале у вас не должно быть такого, чтобы падало с чексуммой. Для этого не нужно изменять существующие миграции. Как вы тогда будете проблему решать после деплоя?

shinehlina replied:
Так что вам не нужна эта строчка

Murwar replied:
@shinehlina 
Это на время проектирования БД, так как часто менялась структура, я решила проблему именно так, потому что создавать каждый раз новую версию немного захламительно. У ребят тоже была проблема с flyway в какой-то момент, а clean, repair не помогали, так что написала и внесла данную строчку в общий код именно для этого. 

Далее же будут вноситься изменения с помощью новых версий миграции, а строчка будет закоменчена.

DeLonghi committed:
Delete .factorypath

DeLonghi committed:
Closed file input stream

DeLonghi committed:
Version 0.2.0

Anglopol reviewed (approved):
Немного отрефакторить еще бы, но перенесем это на следующий раз :)

Anglopol commented:
Можете мерджить
